tests/test_generic.py

```diff
@@ -472,23 +474,25 @@ def test_TextStringObject_autodetect_utf16():
 
 
 def test_remove_child_not_in_tree():
+    class ChildDummy(DictionaryObject):
+        @property
+        def indirect_ref(self):
+            return self
+
```

Comment:
You could move this class outside of the test. This way you would not have to repeat it in the other tests.

Maybe adding a docstring to the class would then be a good idea:

```
    class ChildDummy(DictionaryObject):
         """A helper class for testing interactions with the TreeObject."""
        @property
        def indirect_ref(self):
            return self
```

Reply:
isolated into #1411